packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -561,8 +572,7 @@ export function createArrayNodeProxy(
 				return Reflect.set(dispatch, key, newValue, proxy);
 			}
 
-			const field = getSequenceField(proxy);
-			const maybeIndex = asIndex(key, field.length);
+			const maybeIndex = asIndex(key, Infinity); // JS arrays allow setting at indices greater than the length
```

Comment:
do we have tests for "holey" arrays?

Reply:
No.  We hadn't planned on supporting "holey" arrays in the MVP.  One reason we were okay with punting "holey" arrays is that they are not supported in JSON.[1]

However, we did intend to allow this non-holey common idiom:

array[array.length] = x;    // Append to end, like `array.push(x)`

[1] JSON coerces holes to null ([0, <1 hole>, 2] -> [0, null, 2]), which might be a schema violation.

Reply:
We also don't allow `undefined` to be a value in an array, even when it's not a "hole". For example, you can't do:
`sf.array(sf.optional(sf.number))`
for the same reasons.